runtime/reconcilers/connector.go

```diff
@@ -61,8 +64,68 @@ func (r *ConnectorReconciler) Reconcile(ctx context.Context, n *runtimev1.Resour
 
 	// Exit early for deletion
 	if self.Meta.DeletedOn != nil {
+		err = r.C.Runtime.UpdateInstanceConnector(ctx, r.C.InstanceID, self.Meta.Name.Name, nil)
+		if err != nil {
+			return runtime.ReconcileResult{Err: err}
+		}
 		return runtime.ReconcileResult{}
 	}
 
+	// Check if the spec has changed
+	specHash, err := r.executionSpecHash(t.Spec)
+	if err != nil {
+		return runtime.ReconcileResult{Err: err}
+	}
+
+	if t.State != nil && specHash == t.State.SpecHash {
+		return runtime.ReconcileResult{}
+	}
+
+	// Update instance connectors
+	err = r.C.Runtime.UpdateInstanceConnector(ctx, r.C.InstanceID, self.Meta.Name.Name, t.Spec)
+	if err != nil {
+		return runtime.ReconcileResult{Err: err}
+	}
+
+	if t.State == nil {
+		t.State = &runtimev1.ConnectorState{}
+	}
```

Comment:
If you update `func (r *ConnectorReconciler) ResetState(res *runtimev1.Resource) error {` to set an empty state, then this isn't needed. We do that in most other reconcilers, but actually looks like there's some inconsistency, so I understand the confusion. There might be some other code that assumes the state is not nil, so it's probably best to set it there.